addon/components/transition.js

```diff
@@ -0,0 +1,72 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { use } from 'ember-could-get-used-to-this';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import {
+  AppliedClassNamesResource,
+  TransitionVisibilityResource,
+} from '../resources/transition';
+
+export default class TransitionComponent extends Component {
+  constructor() {
+    super(...arguments);
+
+    let { show } = this.args;
+
+    assert(ARG_SHOW_MISSING_ERROR_MESSAGE, typeof show === 'boolean');
+
+    // Normally this value is computed lazily, but we *do not want this*
+    // If we allow the value to be lazy, the class names are first computed only once the component is rendered
+    // Simply accessing the property will cause the resource to begin computing
+    this.transitionClassNames;
+  }
+
+  get tagName() {
+    return this.args.tagName ?? 'div';
+  }
+
+  get unmount() {
+    return this.args.unmount ?? true;
+  }
+
+  /* === Component Visibility Management === */
+
+  @tracked domNodes = new Set();
```

Comment:
This, and the `domNode` on line 63, need better names

This property contains all of the DOM nodes related to the entire `Transition` tree (including children)

The other property is specifically the DOM node for _this_ `Transition` component